tests/Zend/Form/ElementTest.php

```diff
@@ -1760,12 +1765,12 @@ public function testSetOptionsSetsArrayOfArrayFilters()
 
         $options = $this->getOptions();
         $options['filters'] = array(
-            array('Digits', array('bar' => 'baz')),
+            array('Alnum', array('allowWhiteSpace' => true)),
             array('Alpha', array('foo')),
         );
         $this->element->setOptions($options);
-        $filter = $this->element->getFilter('Digits');
-        $this->assertTrue($filter instanceof Zend_Filter_Digits);
+        $filter = $this->element->getFilter('Alnum');
+        $this->assertTrue($filter instanceof Zend_Filter_Alnum);
```

Comment:
the behavior of this tests has changed. Could you comment on why, please?

Reply:
The behaviour hasn't really changed because the intent of the test is to use an array of arrays in `setOptions`. What has changed is that PHP8 supports calling functions with named parameters and will throw a Fatal if you try to access a parameter by name that does not exist. `Digits::__construct` has no param named bar ergo we get a Fatal.

I could add a version switch but that wouldn't make the test any more meaningful.